### ISSUE TITLE
Implement Lab Journal feature

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#208]
***In GitLab by @wiglef on Sep 5, 2024, 13:40 GMT+2:***

# Feature Request

## :goal: Goal

As a `researcher at DLR`I want `to journal my experiments, lab and project work in a way that is connected to the other research data` so that `me and other researchers can revisit and understand my lab work, experiments and corresponding data at a later point in time`

## :medal: Acceptance Criteria

Updated Link to new Design:
https://www.figma.com/design/ivBPq71AuUl2ohFnrzHIKF/Visual-Design?node-id=67-14534&node-type=frame&m=dev

1. There is a Journal feature that allows to enter journal entries for all data objects
2. Journal entries can be entered in plain text
3. Text formatting options are available 
   1. Hyperlinks are also a desired text formatting option.
   2. Tables feature from editor.js shall also be included.
4. Journal entries have an author that is displayed with the entry
5. Permission concept: 
- View, Edit, Change and Delete permission are defined by the data object 

6. Edited by remark: If an existing lab journal entry has been edited, a little remark is appended at the bottom of the journal entry with editor user name & date for the latest update. 

## :bomb: Out of Scope

* a feature to write direct short links to data objects or users (like with @ or \~ in JIRA) is an idea for later
* a feature to write project (collection) wide journal entries "announcements" might be tested with users later
* the functonality to paste images directly to the text will be done in a follow up story
* There is a Lab Journal overview page on collection level that displays a list of all journal entries in the collection, ordered by date

## :bulb: Hints

* Selection of a standard library for text editors is desired. First idea is https://editorjs.io/ but that's not set in stone
  * Storage of images to text may require to retrieve images direct from URL, which we do not have in the moment
* Solution ideas: 
  * It is desired to implement the storage inside the existing data model to neatly support search, export and API functionalities
* 

## :thinking: Risks & Open Questions

Some guidance to the user would be helpful to make them understand what fits better in a description and what should be a journal entry

**Assignees:** xit-eyad-aljawabreh

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/208*

**Comment by noheton** [comment]
***In GitLab by @wiglef on Sep 24, 2024, 15:07 GMT+2:***

marked this issue as blocked by #247

**Comment by noheton** [comment]
***In GitLab by @wiglef on Sep 24, 2024, 15:07 GMT+2:***

marked this issue as blocked by #246

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 11, 2024, 15:55 GMT+1:***

added dlr-shepard#16 as parent epic

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 13, 2024, 15:29 GMT+1:***

marked #246 as a duplicate of this issue

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 13, 2024, 15:29 GMT+1:***

removed the relation with #247

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 13, 2024, 15:29 GMT+1:***

removed the relation with #246

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 19, 2024, 16:56 GMT+1:***

added #317 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 21, 2024, 09:51 GMT+1:***

Assumptions during Detail Planning 21.11.:
- In the data object, we can click "+" to add a lab journal entry
- We don't create lab journal entries in the collection list for now
- Lab journal entries are created & edited in an inline form
- Out of Scope: Deletion of lab journal entries

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Nov 21, 2024, 10:36 GMT+1:***

added #321 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Nov 21, 2024, 10:42 GMT+1:***

added #322 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Nov 21, 2024, 10:46 GMT+1:***

added #323 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Nov 21, 2024, 10:48 GMT+1:***

added #324 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Nov 21, 2024, 10:55 GMT+1:***

added #326 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 21, 2024, 15:01 GMT+1:***

removed child task #321

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 21, 2024, 15:02 GMT+1:***

Decision 21.11. by @wiglef : We postpone the collection lab journal overview until we have the collection detail page.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 25, 2024, 11:01 GMT+1:***

added #331 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 25, 2024, 11:01 GMT+1:***

mentioned in task #331

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 27, 2024, 10:05 GMT+1:***

@xit-paul-reinhardt @ZeroOne42  

I clarified the permission AC in step 5. I do think that it makes sense this way. 

However I also see the point that we could reduce complexity if we'd just get rid of lab journal specific permissions and inherit everything from data object, so I'll use the concept weekly to discuss if we could just do that, at least for the first version.

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 27, 2024, 10:13 GMT+1:***

gitlab, for instance, also just lets me edit Pauls comments so that I can write something stupid in his name, but at least its displaying a little change note. 

undefined

i would also be fine with that if we get that little change note. i think users in our context are generally aware that they shouldn't mess with other peoples documentation. i'll come back to you after concept weekly

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 27, 2024, 11:20 GMT+1:***

we just talked it in the concept weekly. we decided to reduce complexity and not have seperate permissions, but display a little update remark if lab journal has been updated.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 29, 2024, 09:21 GMT+1:***

marked this issue as blocking #343

**Comment by noheton** [comment]
***In GitLab by @wiglef on Dec 3, 2024, 15:23 GMT+1:***

added #348 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Dec 4, 2024, 13:11 GMT+1:***

added #349 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Dec 4, 2024, 13:22 GMT+1:***

marked this issue as blocking #315

**Comment by noheton** [comment]
***In GitLab by @wiglef on Dec 6, 2024, 10:17 GMT+1:***

added #350 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Dec 9, 2024, 12:48 GMT+1:***

mentioned in commit f03f7687a702f4eebad2e952faad4782355c4a5c

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Dec 9, 2024, 13:55 GMT+1:***

mentioned in commit 2ce81e72d880c292a3bda58129cf86b340d541b9

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Dec 9, 2024, 15:02 GMT+1:***

mentioned in commit 09de5b2dfc21795540304cc34e1061caac53dbc7

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Dec 9, 2024, 15:17 GMT+1:***

mentioned in commit 4619566eba6dc68b762fa37a00724040f84f7205

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Feb 25, 2025, 09:03 GMT+1:***

marked this issue as related to #508
